### ISSUE TITLE
not used right now

### ISSUE THREAD

**Issue by michaelbekos** [michaelbekos/crossingresolution#33]
1. What is the usage of this slider? If I change its value on a graph with n=400, e=1600 the drawing algorithm is crashed.

2. I do not see any repulsive forces in your algorithm. In a sense the graph is being shrink at each iteration. Are you sure that this modification you made at the repulsive forces has an effect? I also see that the min angle is getting smaller, isn't it?

**Comment by plneappl** [comment]
1. The slider was used to adjust the spring length. However, we now set the length based on the number of nodes in graphs. Changing it is undefined behaviour, but since this is an internal tool, I don't think that's an issue.
2. When reporting bugs, please specify exact steps to reproduce them :) i.e. upload a graphml file or sth., It seems fine on my end.

**Comment by michaelbekos** [comment]
1. I do not understand this reply. If the slider was used to adjust the spring length and right now it doesn't, what's the reason of having the slider there? 

2. I didn't report a bug. Try any graph yourself (that is not small). You will see that any two vertices are getting at each iteration closer and closer to each other and there is no way to apply some kind of repulsive force to make the drawing bigger. Unless you start playing with the first slider, which has this upper limit that I reported in my other comment. That's what I reported and my question was clear: "Are you sure that this modification you made at the repulsive forces has an effect?"

**Comment by plneappl** [comment]
1. Not doing unneccessary work :) Especially since one might want to introduce it later, and since it's quite a hassle with the current code to remove it, as its automatically generated.
2. It looked like a bug report to me... Anyhow,  it works as expected™, that is, very dense graphs have a lot of pulling springs. Very sparse graphs however get pushed apart.

**Comment by michaelbekos** [comment]
To make the story short, here is the problem: 

I worked during the weekend on the tool. In most of the graphs I tried to improve the crossing resolution I failed. The reason was that I had no way to really play with the drawing. First, the sliders were of limited size (so, I couldn't really take advantage of them). 

On the other hand, I had no repulsive forces to make the drawing bigger; it was always small. For the competition that's limitation. 

So, my suggestion is the following: Bring back the repulsive forces as an option, that is, I would like to choose whether to you use the variant that you currently have implemented or the standard repulsive forces (together with their slider to be able to control). 

This change must come together with the change on the sliders that I suggested on the other issue.
